compiler/quilt/test/test_command.py

```diff
@@ -110,6 +110,34 @@ def test_version_add_declined(self, mock_input):
         mock_input.return_value = 'n'
         command.version_add('user/test', '2.9.12', 'fabc123')  # should produce no mock network activity
 
+    def test_ambiguous_hash(self):
+        registry_url = command.get_registry_url()
+        session = command._get_session()
+        ambiguous_token = "795a7b"
+        # There should be at least two results that start with the ambiguous_token, plus some non-ambiguous
+        # results in fake_data to test against.
+        fake_data = {'logs': [
+            {'author': 'user', 'created': 1490816524.0,
+             'hash': '885696c6e40613b3c601e95037caf4e43bda58c39f67ab5d5e56beefb3662ff4'},
+            {'author': 'user', 'created': 1490816507.0,
+             'hash': '795a7bc9e40613b3c601e95037caf4e43bda58c39f67ab5d5e56beefb3662ff4'},
+            {'author': 'user', 'created': 1490816473.0,
+             'hash': '795a7bc6e40613b3c601e95037caf4e43bda58c39f67ab5d5e56beefb3662ff4'},
+            {'author': 'user', 'created': 1490816524.0,
+             'hash': '2501a6c6e40a7b355901fc5037caf4e43bda58c39f67ab5d5e56beefb3662ff4'},
+        ]}
+        self.requests_mock.add(
+            responses.GET,
+            registry_url + "/api/log/user/test/",
+            json=fake_data
+        )
+        # Ambiguous hashes in _match_hash's exception will be sorted -- sorted here to match.
+        fake_data_ambiguous = sorted(entry['hash'] for entry in fake_data['logs']
+                               if entry['hash'].startswith(ambiguous_token))
+        fake_data_regexp = '[\s\S]'.join(fake_data_ambiguous)
```

Comment:
Why `[\s\S]`? Isn't it just `.`?

Reply:
Newlines in the exception.

Reply:
Commented explaining, per @asah.  I also made the match more flexible, in case the CommandException's text changes.